packages/cozy-viewer/src/ViewersByFile/PdfMobileViewer.jsx

```diff
@@ -32,36 +33,42 @@ export const PdfMobileViewer = ({ file, url, t, gestures }) => {
     setLoading(false)
   }
 
-  const handleOnClick = useCallback(
-    async file => {
-      try {
-        await downloadFile({ client, file, webviewIntent })
-      } catch (error) {
-        showAlert({
-          message: t('Viewer.error.generic'),
-          severity: 'error',
-          variant: 'filled',
-          icon: false
-        })
-      }
-    },
+  const handleOnClick = useMemo(
+    () =>
+      throttle(
+        async file => {
+          try {
+            await downloadFile({ client, file, webviewIntent })
+          } catch (error) {
+            showAlert({
+              message: t('Viewer.error.generic'),
+              severity: 'error',
+              variant: 'filled',
+              icon: false
+            })
+          }
+        },
+        1000,
+        { trailing: false }
+      ),
```

Comment:
info: This approach is also functional
https://github.com/cozy/cozy-contacts/blob/26b30f5160ff6e7543e7f2e134bd1893f701c99b/src/components/Modals/ContactInfoTitle.jsx#L62-L75

Reply:
Interesting thanks 👍